lib/push.py

```diff
@@ -87,6 +90,11 @@ def cleanup(self):
          ch.VERBOSE("deleting tarball: %s" % tar_c)
          ch.unlink(tar_c)
 
+   def hash_only(self):
+      self.prepare()
+      for (i, (digest, tarball)) in enumerate(self.layers, start=1):
+         ch.INFO("layer %d/%d: %s" % (i, len(self.layers), digest[0:7]))
+
```

Comment:
IMO this bail-out should go in `push()`, to avoid creating another code path.

Also, I see that in this mode, the cleanup doesn't happen. That seems like a good idea, but it's not clear it was a deliberate choice.